### PR TITLE
Urgent fix to show user location on Task Manager

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3828

### Change description ###
Request to reduce the backend load by ensuring that any user will be shown the Task Manager page with the location filter defaulted to their base location (where possible).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by udaydenduluri33 — DISMISSED** [review]
LGTMe

**Review by leepavlouhmcts — APPROVED** [review]
approved

**Review by udaydenduluri33 — APPROVED** [review]
LGTM after the merge from master

File: src/work-allocation/components/task-manager-filter/task-manager-filter.component.ts [hunk 1]
```diff
@@ -87,11 +91,12 @@ export class TaskManagerFilterComponent implements OnChanges {
     this.sessionStorageService.setItem(FilterConstants.Session.TaskManager, toStore);
   }
 
-  public getLocationById(id: string): Location {
+  public getLocationById(id: any): Location {
```
Comment (ronaldmansveld) [inline src/work-allocation/components/task-manager-filter/task-manager-filter.component.ts:94]: Should never use any. Either make changes in the rest of the code so it's always a string, or, if that can't be done, make this a compound type.

File: src/work-allocation/containers/task-manager-list/task-manager-list.component.ts [hunk 2]
```diff
@@ -43,6 +45,25 @@ export class TaskManagerListComponent extends TaskListWrapperComponent implement
...
+   * Gets the caseworker's location
+   */
+  public get caseworkerLocation(): Location {
+    return this.pCaseworkerLocation;
+  }
+
+  private pUserId: string;
+  // pCaseworkerLocation is the caseworker that sets the location of the location dropdown
+  // Note: Setter for caseworkerLocation may come in useful if the selected location needs to be set via the caseworker assigned to the task
+  public pCaseworkerLocation: Location;
```
Comment (ronaldmansveld) [inline src/work-allocation/containers/task-manager-list/task-manager-list.component.ts:65]: Why the prefix 'p' for this public property?

**Review by ronaldmansveld — CHANGES_REQUESTED** [review]

**Review by leepavlouhmcts — APPROVED** [review]

File: src/work-allocation/containers/task-manager-list/task-manager-list.component.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/work-allocation/containers/task-manager-list/task-manager-list.component.ts]: fixed the issues

File: src/work-allocation/components/task-manager-filter/task-manager-filter.component.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/work-allocation/components/task-manager-filter/task-manager-filter.component.ts]: Agreed, changed it to string

File: src/work-allocation/containers/task-manager-list/task-manager-list.component.ts [hunk 3]
```diff
@@ -84,6 +104,40 @@ export class TaskManagerListComponent extends TaskListWrapperComponent implement
...
+
+  /**
+   * Using set user id, gets caseworker details for the caseworker which will set the selected location
+   * (caseworker for the purpose of selecting location in dropdown currently the logged in user)
+   */
+  private setupCaseworkerLocation(): void {
+    this.caseworkerService.getAll().subscribe(caseworkers => {
+      const assignedCaseworker = caseworkers.find(cw => this.isLoggedInUser(cw.idamId));
+      this.pCaseworkerLocation = assignedCaseworker.location ? assignedCaseworker.location : null;
+      // reset location to test - TODO - Remove comments once tested
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply src/work-allocation/containers/task-manager-list/task-manager-list.component.ts]: Done

**Review by leepavlouhmcts — APPROVED** [review]

**Review by ronaldmansveld — APPROVED** [review]

**Pull request merged**